Doc/naming-conventions.md

```diff
@@ -0,0 +1,88 @@
+# Metric and tag naming convention for M6 usage
+
+## 1. Metric naming convention
+
+### Follow Prometheus guideline
+
+A metric should respect this format:
+```
+{source}_{subject/type}_({direction:in/out}_)_{unit}
+```
+
+1. A __source__ prefix: `http`, `worker`, `command`, etc.
+2. An explicit __subject__: `request`, `execution_duration`
+3. A __direction__: does the metric monitore something coming `in`,  `out` or is internal?
+4. And a __unit__ suffix: 
+    * increment => `_total`
+    * counter => `_count`
+    * duration => `_seconds`
+    * memory => `_bytes`
+    * etc.
+
+Please, have a look at the [Prometheus metric and label naming convention](https://prometheus.io/docs/practices/naming/).
+
+### Allowed characters and format
+
+In order to name the metric, you can use lower case letters, numbers and underscores: __[a-z0-9\_]__.
+
+You are recommended to name the metrics in __snake case__. 
+
+The statsd_exporter will transform every of metric names in snake_case anyway. 
+It will be easier for you to find your metrics later if the names are the same. 
+
+
+### Unity
+
+Metrics name must be __unique__. 
+
+You cannot have the same metric name, with a different type or differents tag names, for example.
```

Comment:
Does this mean that the following example wont work ? :fearful: 

**In Project A**
Metric `request_toto_seconds` with tags `project:ProjectA`, `route:{random_route_value}`

**In Project B**
Metric `request_toto_seconds` with tags `project:ProjectB`, `route:{random_route_value}`, `client:{random_client_value}`

Reply:
You understood it right. 
In this example, if the metric from Project A is sent first, then, when Project B will send its metric, the statsd_exporter will ignore the tag "client" until you restart the server. After the server has been restarted, if Project A begins, you'll have the same trouble endlessly...

That's why we are working on a naming convention. having "normalized" tags would prevent this from happening.

Reply:
I'll replace "metric name must be unique" with "metric configuration must be unique" and explicit what "configuration" means (name, tags, type), it would be easier to understand.